### PR TITLE
Monitoring: Explicitly name widget transactions

### PR DESCRIPTION
**Pull request by SarahJaine**
## Description of change
We wanted to have an idea of what the user experience (response time and throughput) is with the widgets before we release the dashboard features to the main user group. Currently, the dashboard is only available to admin users.

Unfortunately New Relic isn't playing nicely with our node js set up, and the api routes are being grouped. Under APM > Transactions > you'll see the widget traces (and a lot of other api calls) appear under `/*` transaction. 

This PR uses the new relic api to name the widget transactions so we can see their performance in new relic.

## How to test
Check out sandbox in new relic. The screen grabs of the transaction pages are from sandbox, which contains only fake data.
Here's a [link](https://one.newrelic.com/launcher/nr1-core.explorer?platform[accountId]=3062577&platform[filters]=Iihkb21haW4gPSAnQVBNJyBBTkQgdHlwZSA9ICdBUFBMSUNBVElPTicpIg==&platform[timeRange][begin_time]=1628175900000&platform[timeRange][end_time]=1628177280000&platform[$isFallbackTimeRange]=false&pane=eyJuZXJkbGV0SWQiOiJhcG0tbmVyZGxldHMudHJhbnNhY3Rpb25zIiwiZW50aXR5R3VpZCI6Ik16QTJNalUzTjN4QlVFMThRVkJRVEVsRFFWUkpUMDU4T0RZM016UTJOems1In0=&sidebars[0]=eyJuZXJkbGV0SWQiOiJucjEtY29yZS5hY3Rpb25zIiwic2VsZWN0ZWROZXJkbGV0Ijp7Im5lcmRsZXRJZCI6ImFwbS1uZXJkbGV0cy50cmFuc2FjdGlvbnMifSwiZW50aXR5R3VpZCI6Ik16QTJNalUzTjN4QlVFMThRVkJRVEVsRFFWUkpUMDU4T0RZM016UTJOems1In0=) to an appropriate timeframe of data in new relic. 

**Before** 
Screen Shot 2021-08-05 at 10 15 31 AM

**After**
Screen Shot 2021-08-05 at 10 19 57 AM

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-268

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/routes/widgets/index.js [hunk 1]
```diff
@@ -1,8 +1,15 @@
 import express from 'express';
 import { getWidget } from './handlers';
 
+const nr = require('newrelic');
+
 const router = express.Router();
 
-router.get('/:widgetId', getWidget);
+function nameWidgetTransactions(req, res, next) {
+  nr.setTransactionName(`GET /api/widgets/${req.params.widgetId}`);
```
Comment (AdamAdHocTeam) [inline src/routes/widgets/index.js:9]: Not sure if it matters or is possible, but might be nice to have the name of the widget in addition to the ID.

Comment (SarahJaine) [reply src/routes/widgets/index.js:9]: I had this concern too, but it turns out the widgetId is a good name, not an integer or something inscrutable. Example: `topicFrequencyGraph`.

Comment (AdamAdHocTeam) [reply src/routes/widgets/index.js:9]: Ok perfect.

**Review by jasalisbury — APPROVED** [review]
:shipit:

**Review by AdamAdHocTeam — APPROVED** [review]
I now see the individual widgets and their corresponding txn data in new relic for sandbox.

**Pull request merged**